### PR TITLE
`e2e`: Add test of `platform.getValidatorsAt` across nodes

### PR DESCRIPTION
**Pull request by marun**

### CODE REVIEW

File: tests/e2e/p/validator_sets.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+		ginkgo.By("checking that validator sets are equal across all heights for all nodes", func() {
+			pvmClients := make([]platformvm.Client, len(e2e.Env.URIs))
+			for i, nodeURI := range e2e.Env.URIs {
+				pvmClients[i] = platformvm.NewClient(nodeURI.URI)
+			}
+
+			initialHeight, err := pvmClients[0].GetHeight(e2e.DefaultContext())
+			require.NoError(err)
+
+			for height := initialHeight; height > 0; height-- {
```
Comment (StephenButtolph) [inline tests/e2e/p/validator_sets.go:80]: I think we can check this for `height == 0` too right? (watchout for underflow, might make sense to iterate from `height=0` to `height <= initialHeight`)

File: tests/e2e/p/validator_sets.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+						height,
+					)
+					require.NoError(err)
+					if observedValidatorSet == nil {
+						observedValidatorSet = validatorSet
+						continue
+					}
+					require.Equal(observedValidatorSet, validatorSet)
+				}
+			}
```
Comment (StephenButtolph) [inline tests/e2e/p/validator_sets.go:96]: I think this is probably going to be flaky. We get the height from `pvmClients[0]` - but the other nodes may be behind right?

also `pvmClients[0]` itself could be behind. I think we probably want to get the height from the `nodeURI` used by the wallet -> then when we're iterating over the heights, we should make sure the node has advanced to the expected height before calling `GetValidatorsAt`

File: tests/e2e/p/validator_sets.go [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+		})
+
+		ginkgo.By("checking that validator sets are equal across all heights for all nodes", func() {
+			pvmClients := make([]platformvm.Client, len(e2e.Env.URIs))
+			for i, nodeURI := range e2e.Env.URIs {
+				pvmClients[i] = platformvm.NewClient(nodeURI.URI)
+			}
+
+			initialHeight, err := pvmClients[0].GetHeight(e2e.DefaultContext())
```
Comment (StephenButtolph) [inline tests/e2e/p/validator_sets.go:77]: `initialHeight` seems a bit odd - perhaps `currentPChainHeight` or something?

File: tests/e2e/p/validator_sets.go [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+		pWallet := baseWallet.P()
+
+		const delegatorCount = 15
+		ginkgo.By(fmt.Sprintf("adding %d delegators", delegatorCount), func() {
+			rewardKey, err := secp256k1.NewPrivateKey()
+			require.NoError(err)
+			avaxAssetID := pWallet.AVAXAssetID()
+			startTime := time.Now().Add(tmpnet.DefaultValidatorStartTimeDiff)
+			endTime := startTime.Add(time.Second * 360)
+			// TODO(marun) Ensure this is appropriate for the targeted network (is it accessible from the API?)
```
Comment (StephenButtolph) [inline tests/e2e/p/validator_sets.go:46]: Yeah this is fine. The default min delegator stake is based on the `LocalParams` ([ref](https://github.com/ava-labs/avalanchego/blob/master/config/flags.go#L280))

File: tests/e2e/p/validator_sets.go [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+					e2e.Eventually(
+						func() bool {
+							pChainHeight, err := pvmClient.GetHeight(e2e.DefaultContext())
+							require.NoError(err)
+							return pChainHeight >= currentPChainHeight
+						},
+						e2e.DefaultTimeout,
+						e2e.DefaultPollingInterval,
+						fmt.Sprintf("failed to see expected height %d for %s before timeout", height, e2e.Env.URIs[i].NodeID),
+					)
```
Comment (StephenButtolph) [inline tests/e2e/p/validator_sets.go:95]: I think we can probably move this check out of the `height` loop. We only need to make sure each node has advanced far enough once.

**Review by StephenButtolph — COMMENTED** [review]
After this nit this LGTM. Thanks for putting this up!

**Review by StephenButtolph — APPROVED** [review]

**Review by dhrubabasu — APPROVED** [review]

**Pull request merged**